test/cypress/integration/home_spec.js

```diff
@@ -0,0 +1,93 @@
+describe("Home Page before login", () => {
+  describe("Before login", () => {
+
+    beforeEach(() => {
+        cy.visit("/");
+    });
+
+    it("has a nav bar", () => {
+        cy.get("nav.navbar").should("exist");
+    });
+
+    it("has a Home button", () => {
+        cy.get('a[href="/"]')
+        .should('have.text', 'Home')
+    });
+  }); // before login
+
+  describe("Student login", () => {
+    beforeEach(() => { 
+      cy.request("/api/testhooks/login_student");
+      cy.visit("/");
+    });
+
+    it("has a nav bar", () => {
+        cy.get("nav.navbar").should("exist");
+    });
+
+    it("has a Home button", () => {
+        cy.get('a[href="/"]')
+        .should('have.text', 'Home')
+    });
+
+    it("has a Courses button", () => {
+        cy.get('a[href="/courses"]')
+        .should('contain.text', 'Courses')
+    });
+  }); // Student login
+
+  describe("Admin login", () => {
+
+    let courseUrl;
+
+    beforeEach(() => { 
+        cy.request("/api/testhooks/login_admin");
+        cy.visit("/courses");
+        cy.get('table#course_list')
+            .find('a')
+            .invoke('attr', 'href')
+            .then((href) => {
+                cy.log(`href=${href}`);
+                courseUrl = href
+            }).then( () => {
+                cy.visit(courseUrl)
+            });
```

Comment:
Another potential place to use `cy.click`

Reply:
Not unless cy.click can capture the url value and store it in a variable.  That's what this code is doing.